src/current/cockroachcloud/upgrade-to-v24.2.md

```diff
@@ -0,0 +1,296 @@
+    {% capture red_date %}{{ eos | minus: 2592000 }} {% endcapture %}{% comment %}1 month in seconds{% endcomment %}
+  {% endif %}
+
+  {% if today < eos %}
+    {% assign supported = true %}
+    {% assign red_result = today | minus: red_date %}
+    {% assign purple_result = today | minus: purple_date %}
+  {% endif %}
+{% endif %}
+
+{% if DEBUG == true %}
+skippable: {{ skippable }}<br />
+supported: {{ supported }}<br />
+today: {{ today }}<br />
+eos: {{ eos }}<br />
+eos_date: {{ eos_date }}<br />
+purple_date: {{ purple_date }}<br />
+red_date: {{ red_date }}<br /><br />
+purple_result: {{ purple_result }}<br /><br />
+red_result: {{ red_result }}<br />
+{% endif %}
+
+{% capture testing_release_message %}
+{{site.data.alerts.callout_info}}
+CockroachDB {{ x.major_version }} is a testing release{% if page.pre_production_preview == false %} and is not yet available in CockroachDB {{ site.data.products.cloud }}{% endif %}. [Testing releases]({% link releases/index.md %}#release-naming) are not qualified for production environments and not eligible for support or uptime SLA commitments. This page is provided for reference only and is subject to change.
+{{site.data.alerts.end}}
+{% endcapture %}
+
+{% capture support_ending_soon %}
+Support for CockroachDB v24.2 will end on **{{ eos_date }}** at the end of its Maintenance Support phase.{% if skippable == true %} Optional innovation releases are not eligible for Assistance Support.{% endif %} Prior to that date, upgrade to a newer version. For more details, refer to [CockroachDB Cloud Upgrade Policy]({% link cockroachcloud/upgrade-policy.md %}).
+{% endcapture %}
+
+{% capture eos_message %}
+{{site.data.alerts.callout_danger}}
+As of **{{ eos_date }}**, CockroachDB {{ x.major_version }} is no longer supported. For more details, refer to [CockroachDB Cloud Upgrade Policy]({% link cockroachcloud/upgrade-policy.md %}). This page is provided for reference purposes only.
+{{site.data.alerts.end}}
+{% endcapture %}
+
+{% if released == true %}
+  {% if supported == false %}
+    {{ eos_message }}
+  {% else %}
+{% elsif released %}
+
+    {% if red_result < eos %}
+{{site.data.alerts.callout_danger}}
+{{ support_ending_soon }}
+{{site.data.alerts.end}}
+    {% elsif purple_result < eos %}
+{{site.data.alerts.callout_version}}
+{{ support_ending_soon }}
+{{site.data.alerts.end}}
+    {% endif %}
+  {% endif %}
+{% else %}
+{{ testing_release_message }}
+{% endif %}
+
+{% capture previous_version_numeric %}{{ x.major_version | remove_first: 'v' }}{% endcapture %}
+{% capture major_version_numeric %}{{ x.previous_version | remove_first: 'v' }}{% endcapture %}
+
+{% if DEBUG == true %}
+previous_version_numeric: {{ previous_version_numeric }}<br />
+major_version_numeric: {{ major_version_numeric }}<br />
+{% endif %}
+
+{% if page.pre_production_preview == true and released == false %}
```

Comment:
All changes above this are tooling and I don't expect a review, just verify that the preview looks correct.

Reply:
tbh I can't even tell if the preview is correct or not but let's fix that in another PR if needed